sympy/functions/combinatorial/factorials.py

```diff
@@ -618,8 +625,10 @@ def eval(cls, x, k):
                 return S.Zero
 
     def _eval_rewrite_as_gamma(self, x, k, **kwargs):
-        from sympy import gamma
-        return gamma(x + k) / gamma(x)
+        from sympy import gamma, Piecewise
+        return Piecewise(
+            (gamma(x + k)/gamma(x), x > 0),
+            ((-1)**k*gamma(1 - x)/gamma(-k - x + 1), True))
```

Comment:
I'd suggest you look at the ramifications of including this in here ...
you've skipped a lot of nuances by setting the second option to `True`
the options here are simply not one or the other ... 
take the instance of 
```python3
(x, k) = (-1, 2.5)
```
[`rf(-1, 2.5) = 0`](https://www.wolframalpha.com/input/?i=rising+factorial+-1%2C2.5)
but in your direct substitution, you'd get something like ...
```python3
(-1)**k*gamma(1 - x)/gamma(-k - x + 1)
>>> _.subs({x:-1, k:2.5})
-0.282094791773878*I
```
which is incorrect

Reply:
Note that `k` must be an integer.

Reply:
Yeahhh ... I remember making such accommodations here 
https://github.com/sympy/sympy/blob/9ce3312158638b8fb926100dba03303ec6102c11/sympy/functions/combinatorial/factorials.py#L614-L621